### DISCUSSION TITLE
Cannot segment the robot in the real world? Why the pointcloud is wrong?

### DISCUSSION THREAD

**Discussion by lovelyppp** [NVlabs/curobo#395] [Software Q&A] [answered]
My real world setup contains UR3, robotiq 2F-85 and realsense D435i. When i use the demo robot_image_segmentation_example.py with my own urdf and yml, it succeed as following pictures.
The camera_depth:
![camera_depth](https://github.com/user-attachments/assets/98d17a6a-9219-4bdd-a1df-cdb2273f0da3)
The robot_depth:
![robot_depth](https://github.com/user-attachments/assets/5ff9ca97-06a3-4764-a81e-b7f66a413c15)
However when i modify this python file and apply it into the real world, it doesn't work.
The camera_depth:
![camera_depth](https://github.com/user-attachments/assets/2a2bed88-0d72-4b7e-9005-897680459ce7)
The robot_depth:
![robot_depth](https://github.com/user-attachments/assets/34493c0c-7dd6-47ce-91f4-71388380db63)
And i found that the camera_pointcloud is totally wrong in respect of size and  position information:
![snapshot00](https://github.com/user-attachments/assets/3f25ff65-399b-41aa-a18f-9d01f41f807d)
and here is the size comparation between the robot_spheres and the camera_pointcloud:
![snapshot01](https://github.com/user-attachments/assets/1fb9b882-44a7-45f8-9a36-63cbbb41190c)
The following is my code link:
[curobo_seg](https://github.com/lovelyppp/curobo_seg/blob/main/sdf_reconstruction/robot_image_segmentation_example.py)
I want to know how to fix this problem. Thanks.
@balakumar-s

**Comment by lovelyppp** [comment]
What's more, i calibrate my camera pose by using the ros bag easy_handeye.

**Comment by balakumar-s** [comment, marked as answer]
It could be scale mismatch when projecting the depth to pointcloud. We now have a ros2 package for robot segmentation that could help your use case: https://nvidia-isaac-ros.github.io/repositories_and_packages/isaac_ros_cumotion/isaac_ros_cumotion/index.html#robot-segmentation
